### DISCUSSION TITLE
help with Artisan software

### DISCUSSION THREAD

**Discussion by azcharlie16** [artisan-roaster-scope/artisan#1286] [General]
Just purchased Hottop 2k+ and downloaded Artisan to my PC.  Selected the 2k in config, and set COM3.   When I connect PC to the 2k+, neither the ET, BT or RoR will register.  The timeline moves and chart shows fan increase/decrease and heat increase/decrease.  Can someone tell me how to get the ET, BT and ROR, please?  What am I missing in the configuration.

**Comment by Gagan-Yadav** [comment]
Check your thermocouples:
Make sure the thermocouples (temperature probes) are properly inserted into the Hottop 2K+ roaster. Verify that they are securely connected and in the correct ports. ET and BT are typically connected to different thermocouples, so double-check their placements.

Check your connections:
Ensure that the thermocouple wires are securely connected to the Hottop 2K+ and your PC. Loose connections can result in temperature readings not registering.

Verify COM Port:
Double-check that you've selected the correct COM port (COM3 in your case) in the Artisan configuration. You can check your computer's device manager to confirm that the Hottop 2K+ is recognized and connected to COM3.

Baud Rate and Parity Settings:
In some cases, you might need to adjust the baud rate and parity settings in the Artisan configuration to match the settings of your Hottop 2K+. Ensure that these settings are correctly configured.

Check Software Settings:
Review the settings within the Hottop 2K+ itself. Ensure it's set up to send temperature data over the COM port.

Update Artisan:
Make sure you're using the latest version of Artisan. Sometimes, updating the software can resolve compatibility issues.

Restart Everything:
Sometimes, a simple reboot of your PC and the roaster can resolve connection issues.

**Comment by azcharlie16** [comment]
Thank you so much.   I will recheck everything.

    On Friday, October 27, 2023 at 01:53:04 AM MST, Gagan Yadav ***@***.***> wrote:  
 
 


Check your thermocouples:
Make sure the thermocouples (temperature probes) are properly inserted into the Hottop 2K+ roaster. Verify that they are securely connected and in the correct ports. ET and BT are typically connected to different thermocouples, so double-check their placements.

Check your connections:
Ensure that the thermocouple wires are securely connected to the Hottop 2K+ and your PC. Loose connections can result in temperature readings not registering.

Verify COM Port:
Double-check that you've selected the correct COM port (COM3 in your case) in the Artisan configuration. You can check your computer's device manager to confirm that the Hottop 2K+ is recognized and connected to COM3.

Baud Rate and Parity Settings:
In some cases, you might need to adjust the baud rate and parity settings in the Artisan configuration to match the settings of your Hottop 2K+. Ensure that these settings are correctly configured.

Check Software Settings:
Review the settings within the Hottop 2K+ itself. Ensure it's set up to send temperature data over the COM port.

Update Artisan:
Make sure you're using the latest version of Artisan. Sometimes, updating the software can resolve compatibility issues.

Restart Everything:
Sometimes, a simple reboot of your PC and the roaster can resolve connection issues.

**Comment by MtnBubba** [comment]
I ran into some similar problems setting up Artisan 2.8.4 on a new computer to control my Hottop 2K+., I've been using Artisan with the Hottop for several years on three other computers. Hottop setup is pretty painful. There is an eight year old blog post about how to do it here, https://artisan-roasterscope.blogspot.com/2015/05/hottop-kn-8828b-2k.html, but it's no longer super-accurate. For one thing, it refers to option pages that no longer exist and to options that have been renamed. Once you follow the steps, you'll probably find that the features you want are now working, but you'll find some other things that aren't. 

One quirk I've found is that the heater doesn't shut off when you press the Drop button and the load doesn't completely drop. You can manually turn the heater off with the slider and manually release the load, but neither is ideal. I have automation set up that does everything I need to complete a roast when I'm distracted but I don't remember how I set it all up. 

It would be nice if someone with knowledge of the inner workings of Artisan and Hottop could update the blog post.

**Comment by MAKOMO** [comment]
> I ran into some similar problems setting up Artisan 2.8.4 on a new computer to control my Hottop 2K+., I've been using Artisan with the Hottop for several years on three other computers. Hottop setup is pretty painful. 

Why? It is just one click, selecting the Hottop 2k+ setup from the Config >> Machines menu and selecting the serial port from the popup (while the Hottop is connected and turned on of course as otherwise the OS is not creating that serial port).
 
> There is an eight year old blog post about how to do it here, https://artisan-roasterscope.blogspot.com/2015/05/hottop-kn-8828b-2k.html, but it's no longer super-accurate. For one thing, it refers to option pages that no longer exist and to options that have been renamed. Once you follow the steps, you'll probably find that the features you want are now working, but you'll find some other things that aren't.

That post has for some years a note at the very top that configuration was simplified by the machine menu setup. I just added to this that the roaster should be connected and turned on before running this.

> 
> One quirk I've found is that the heater doesn't shut off when you press the Drop button and the load doesn't completely drop. 

Unclear what an "uncomplete drop" means here. Artisan just instructs the machine to open the drop door.

> You can manually turn the heater off with the slider and manually release the load, but neither is ideal. I have automation set up that does everything I need to complete a roast when I'm distracted but I don't remember how I set it all up.
> 
> It would be nice if someone with knowledge of the inner workings of Artisan and Hottop could update the blog post.

Besides that some UI controls moved around in the dialogs a bit and additional controls have been added the post seems to be still accurate (but not needed any longer as the machine setup does all configuration in one click). What needs to be updated?

Anyhow, there are now plenty of tutorial videos on Artisan and the Hottop. I added the link to two of those to the blog post.

**Reply by MtnBubba** [reply]
Thanks for clarifying that the blog post is no longer needed. I ran into two problems setting up on my new machine: the timer LCD wasn't visible by default, which makes going super user impossible; and the heater isn't turning off when I press DROP at the end of my roast. I figured out how to make the LCD show up, but the heater thing is puzzling.

**Reply by MAKOMO** [reply]
Your heater issue was actually a bug/typo in the Hottop machine setup of the past versions that was never discovered/reported. I found it by accident some days ago and fixed it for the next version.

The DROP action

<img width="925" alt="Screenshot 2023-11-10 at 19 05 09" src="https://github.com/artisan-roaster-scope/artisan/assets/13909011/6292f2cc-da5a-46d8-9201-4ae6e870819e">

should read

```heater(0);fan(10);motor(1);solenoid(1);stirrer(1)```

but was reading

```heater(0);fan(10);drum(1);heater(0);solenoid(1);stirrer(1)````

Note that in this command sequence the `heater(0)` command was duplicated (should not be a problem), but the `drum(1)` command is plain wrong and should read `motor(1)` (also no problem as the motor should already be on at that point and just stay on.

Why your heater is not turned off remains unclear to me. Anyhow, the good news for you and other Hottop users is that the next Artisan version (most likely to be released later this month) will feature a re-implementation of the Hottop communication protocol which should make the communication with the machine more robust.

**Reply by MtnBubba** [reply]
Thank you thank you thank you.

**Reply by MAKOMO** [reply]
Thank you for your generous donation supporting the further development of Artisan! Respect!

**Reply by MtnBubba** [reply]
I did the single-click setup and corrected the drop script and everything is working now. Thanks for the customer service, I need to read more carefully. 

BTW, there's a tool called USBTreeView that's really nice for finding what COMx port the virtual serial driver is using.

**Comment by MtnBubba** [comment]
Your customer service is hard to beat.



*From:* Marko Luther ***@***.***>
*Sent:* Friday, November 10, 2023 12:11 PM
*To:* artisan-roaster-scope/artisan ***@***.***>
*Cc:* Richard Bennett ***@***.***>; Comment <
***@***.***>
*Subject:* Re: [artisan-roaster-scope/artisan] help with Artisan software
(Discussion #1286)



Thank you for your generous donation supporting the further development of
Artisan! Respect!

—
Reply to this email directly, view it on GitHub
<https://github.com/artisan-roaster-scope/artisan/discussions/1286#discussioncomment-7536747>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/ADDEFEKQVL43VIBSNWPUW2TYDZ347AVCNFSM6AAAAAA6QCL4AGVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TKMZWG42DO>
.
You are receiving this because you commented.Message ID: <
***@***.***
>
